src/lib/genesis_ledger_helper/genesis_ledger_helper.ml

```diff
@@ -159,28 +230,63 @@ let retrieve_genesis_state dir_opt ~logger ~conf_dir ~daemon_conf :
       in
       match%bind
         Deferred.List.fold directories ~init:None ~f:(fun acc dir ->
-            if is_some acc then Deferred.return acc else retrieve dir )
+            if is_some acc then Deferred.return acc
+            else retrieve_genesis_data dir )
       with
       | Some res ->
           Deferred.return res
       | None ->
-          (*Check if it's in s3*)
-          let local_path = Cache_dir.s3_install_path ^/ tar_filename in
+          (* Check if genesis data is in s3 *)
+          let tgz_local_path = Cache_dir.s3_install_path ^/ tar_filename in
+          let proof_local_path = Cache_dir.s3_install_path ^/ proof_filename in
+          let constants_local_path =
+            Cache_dir.s3_install_path ^/ constants_filename
+          in
           let%bind () =
             match%map
-              Cache_dir.load_from_s3 [s3_bucket_prefix] [local_path] ~logger
+              Cache_dir.load_from_s3 [s3_bucket_prefix]
+                [tgz_local_path; proof_local_path; constants_local_path]
+                ~logger
             with
             | Ok () ->
                 ()
             | Error e ->
                 Logger.fatal ~module_:__MODULE__ ~location:__LOC__ logger
-                  "Could not curl genesis ledger and genesis proof from $uri: \
-                   $error"
+                  "Could not download genesis ledger, proof, and constants \
+                   from $uri: $error"
                   ~metadata:
                     [ ("uri", `String s3_bucket_prefix)
                     ; ("error", `String (Error.to_string_hum e)) ]
           in
-          let%map res = retrieve Cache_dir.s3_install_path in
+          let%map res = retrieve_genesis_data Cache_dir.s3_install_path in
           res_or_fail
             (String.concat ~sep:"," (s3_bucket_prefix :: directories))
             res )
+
+[%%else]
+
+let retrieve_genesis_proof ~logger : Proof.t Deferred.t =
```

Comment:
Can we also optionally retrieve it from a directory passed as an argument to the daemon (like in consensus nodes)? When running the native exe, we might want to look for the genesis state locally if a directory is specified otherwise download it from s3.

Reply:
Hmm, I wanted to nudge us along the path to Javascript, and require us to use S3. If that proves problematic, we can add the load-from-disk back in (just copying from the consensus code). Is that acceptable?

Reply:
Can we disable the flag for javascript executable later on? The catch is that only medium curve builds and genesis states are uploaded to s3 right now. For all other builds, there is no other way to run non-consensus demons locally or in Ci. Uploading the genesis state for each profile for every commit seems wasteful.

Reply:
Currently, the nonconsensus code profiles all use medium curves. That will change, of course.

I see, our integration tests use small curves, and we'll probably have similar tests for nonconsensus.

I can add the feature, then.